### ISSUE TITLE
Swallowed exception thrown from custom exception handler -> kafka consumer closed

### ISSUE THREAD

**Issue by kziomek** [micronaut-projects/micronaut-kafka#705]
### Expected Behavior

Logging the exception thrown from custom exception handler (which might be a bug) would make it easier to debug and understand what caused closing kafka consumer.

### Actual Behaviour

I had the issue in one of our applications where KafkaConsumer got closed after logging offset commit error.
It wasn't very clear why consumer got closed as I would expect consumer to reconnect the broker.

After debugging and reading micronout kafka code I've found that our custom exception handler throws an exception when ConsumerRecord is null. Exception caused exiting polling loop in _KafkaConsumerProcessor_ without logging the exception.
Then Kafka Consumer got automatically closed.

### Steps To Reproduce

Following listener with custom exception handler reproduces the issue.
Exception thrown from _handle_ method causes exiting infinite polling while loop in _KafkaConsumerProcessor_ and kafka consumer closes. 
"custom handler exception" exception is not logged.

```
@Slf4j
@KafkaListener(
    clientId = "${client-id}",
    groupId = "${group-id}",
    offsetReset = OffsetReset.EARLIEST,
    offsetStrategy = OffsetStrategy.SYNC
public class CustomListener implements KafkaListenerExceptionHandler {
    
    @Blocking
    @Topic("${topic}")
    Completable process(ConsumerRecord<String, String> record) {
        throw new RuntimeException("processing exception");
    }
    
    @Override
    public void handle(KafkaListenerException exception) {
        throw new RuntimeException("custom handler exception");
    }
}
```

### Environment Information

OpenJDK Runtime Environment Temurin-17.0.5+8 (build 17.0.5+8)

### Version

io.micronaut.kafka:micronaut-kafka:4.5.3

**Comment by kziomek** [comment]
I have created PR with a test for this issue. It shows where I believe catching and logging Throwable is useful. 

I tested it locally. However, build in pipelines fails yet before tests have a chance to execute. I thought it might be due to dependency I added for LogCaptor but it fails the same way without changes to dependencies.

Would be great if someone could take a look and help process this PR.
